### PR TITLE
Add jest testing to CI

### PR DESCRIPTION
**Pull request by alfarok**
### Description
This PR addresses #3574.

image

- [x] Add Jest testing framework to project and CI
- [x] Add several test samples covering the new RPC component
- [x] Fix a couple bugs in the RPC component discovered from the new tests

### TODO
- [ ] Add section on Jest testing to README

### CODE REVIEW

File: components/RPC-Connection.jsx [hunk 1]
```diff
@@ -51,10 +51,16 @@ function RPC({ network, path, defaultValue, filter=undefined }) {
...
+				const newValue = await syncData(network, path, setReturnValue);
+				if (newValue === undefined) {
+					 // There was an issue with the request, use default
+					return;
+				} else if (filter !== undefined) {
+					// Apply filter to retrieved value if a filter is provided
+					applyFilter(newValue, filter, network, setReturnValue)
+				} else {
+					// Apply value as-is
+					setReturnValue(newValue);
```
Comment (alfarok) [inline components/RPC-Connection.jsx:63]: This was actually causing an edge case where some 'const' values without filters were always using the default value even though a value was successfully being retrieved from the chain.  I thought the test framework wasn't working at first but it turned out the test was correct.  Unit testing already paying off. :partying_face:

File: package.json [hunk 2]
```diff
@@ -66,8 +72,8 @@
   "dependencies": {
     "@algolia/client-search": "^4.14.1",
     "@mdx-js/runtime": "^1.6.22",
-    "@polkadot/util": "^9.6.2",
-    "@polkadot/util-crypto": "^10.1.1",
+    "@polkadot/util": "10.1.1",
```
Comment (alfarok) [inline package.json:74]: All these dependencies should align in version

File: tests/rpc.test.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import RPC from "../components/RPC-Connection";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+
+// Set max test duration before failing (2 min)
+jest.setTimeout(120000);
+
+test("Retrieves and applies a 'const' RPC value", async () => {
+	render(<RPC network="polkadot" path="consts.balances.existentialDeposit" defaultValue={0} />);
+	await waitFor(() => expect(screen.getByText("10000000000")).toBeInTheDocument(), { timeout: 2000 });
+});
```
Comment (alfarok) [inline tests/rpc.test.js:13]: I want to try loading the entire app and write more generic tests that can potentially find errors in the docs, but I will attempt that in another PR.

**Review by DrW3RK — APPROVED** [review]

**Review by DrW3RK — CHANGES_REQUESTED** [review]
holding this from merging. Please check why this fails

**Comment by alfarok** [comment]
The tests are passing again with no changes.  This makes me think that it is timing related with the delay to receive the RPC response.  I am going to **bump the timeout from 2 seconds to 5 seconds** to see if that helps.  The tests will still complete faster than that if the data is available but wait a little longer if it's not yet.

If any of the tests prove to be more annoying that useful with intermittent failures we can remove or adjust further accordingly.

**Comment by DrW3RK** [comment]
Makes sense. We can merge now

**Review by DrW3RK — APPROVED** [review]

**Comment by alfarok** [comment]
I didn't push the change yet haha but I'll open in a new PR since it's minimal.
